src/hotspot/share/cds/dumpTimeClassInfo.hpp

```diff
@@ -40,32 +40,93 @@ class DumpTimeClassInfo: public CHeapObj<mtClass> {
   bool                         _excluded;
   bool                         _is_early_klass;
   bool                         _has_checked_exclusion;
-public:
-  struct DTLoaderConstraint {
+
+  static void maybe_increment_refcount(Symbol* s) {
+    if (s != NULL) {
+      s->increment_refcount();
+    }
+  }
+  static void maybe_decrement_refcount(Symbol* s) {
+    if (s != NULL) {
+      s->decrement_refcount();
+    }
+  }
+  class DTLoaderConstraint {
     Symbol* _name;
     char _loader_type1;
     char _loader_type2;
+  public:
+    DTLoaderConstraint() : _name(NULL), _loader_type1('0'), _loader_type2('0') {}
     DTLoaderConstraint(Symbol* name, char l1, char l2) : _name(name), _loader_type1(l1), _loader_type2(l2) {
-      _name->increment_refcount();
+      maybe_increment_refcount(_name);
     }
-    DTLoaderConstraint() : _name(NULL), _loader_type1('0'), _loader_type2('0') {}
+    DTLoaderConstraint(const DTLoaderConstraint& src) {
+      _name = src._name;
+      _loader_type1 = src._loader_type1;
+      _loader_type2 = src._loader_type2;
+      maybe_increment_refcount(_name);
+    }
+    DTLoaderConstraint& operator=(DTLoaderConstraint src) {
```

Comment:
Could the src be passed as reference?

Reply:
To use the copy-and-swap idiom, which makes the assignment operator reliably, src needs to be passed by value. See https://stackoverflow.com/questions/3279543/what-is-the-copy-and-swap-idiom